DQM/L1TMonitorClient/src/L1TStage2CaloLayer2DEClient.cc

```diff
@@ -323,12 +420,133 @@ void L1TStage2CaloLayer2DEClient::processHistograms(DQMStore::IGetter &igetter){
+  addObjToSummary(igetter, SummaryHist, std::string("/NonIsolated-EG/NonIsoEGsEta"), "Non-isolated EG iEta");
+  addObjToSummary(igetter, SummaryHist, std::string("/NonIsolated-EG/NonIsoEGsPhi"), "Non-isolated EG iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/Isolated-Tau/IsoTausRank"), "Isolated Tau iEt");
+  addObjToSummary(igetter, SummaryHist, std::string("/Isolated-Tau/IsoTausEta"), "Isolated Tau iEta");
+  addObjToSummary(igetter, SummaryHist, std::string("/Isolated-Tau/IsoTausPhi"), "Isolated Tau iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/NonIsolated-Tau/TausRank"), "Non-isolated Tau iEt");
+  addObjToSummary(igetter, SummaryHist, std::string("/NonIsolated-Tau/TausPhi"), "Non-isolated Tau iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/NonIsolated-Tau/TausEta"), "Non-isolated Tau iEta");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/METRank"), "MET Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/METPhi"), "MET iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/METHFRank"), "METHF Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/METHFPhi"), "METHF iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MHTRank"), "MHT Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MHTPhi"), "MHT iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MHTHFRank"), "MHTHF Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MHTHFPhi"), "MHTHF iPhi");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/ETTRank"), "ETT Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/HTTRank"), "HTT Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MBHFP0"), "MinBiasHFP0");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MBHFM0"), "MinBiasHFM0");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MBHFP1"), "MinBiasHFP1");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/MBHFM1"), "MinBiasHFM1");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/ETTEMRank"), "ETTEM Rank");
+  addObjToSummary(igetter, SummaryHist, std::string("/Energy-Sums/TowCount"), "Tower Count");
+
+  SummaryHist->LabelsOption("v", "x");
+  SummaryHist->LabelsDeflate();
 }
-  
+
+void L1TStage2CaloLayer2DEClient::addObjToSummary(DQMStore::IGetter & getter, TH1F * hist, std::string objPath, const char * binLabel) {
+
+  MonitorElement* dataHist_;
+  MonitorElement* emulHist_;
+
+  dataHist_ = getter.get(input_dir_data_ + objPath);
+  emulHist_ = getter.get(input_dir_emul_ + objPath);
+
+  if (dataHist_ && emulHist_){
+
+    double dataInt = dataHist_->getTH1F()->Integral();
+    double emuInt = emulHist_->getTH1F()->Integral();
+    
+    hist->Fill(binLabel, dataInt/emuInt);
```

Comment:
This code is causing troubles under certain conditions.
At first sight it seems to be caused by division by 0 and the way root handles the rendering of that.
Can you please handle these cases in a clean way? Keep in mind that this code should work for _all_ runs, also non-collision ones without data.
(Not only for this line, but also for the _Divide_s higher up)